kale/embed/gripnet.py

```diff
@@ -241,3 +245,294 @@ def forward(
             z, target_edge_index, edge_type=target_edge_relations, range_list=target_edge_range, if_catout=True
         )
         return z
+
+
+class GripNetInternalModule(Module):
+    r"""
+    The internal module of a supervertex, which is composed of an internal feature layer and multiple internal
+    aggregation layers.
+
+    Args:
+        in_channels (int): the dimension of node features on this supervertex.
+        num_edge_type (int): the number of edge types on this supervertex.
+        if_start_svertex (bool): if this supervertex is a start supervertex on the supergraph.
+        setting (SuperVertexParaSetting): supervertex parameter settings.
+    """
+
+    def __init__(
+        self, in_channels: int, num_edge_type: int, if_start_svertex: bool, setting: SuperVertexParaSetting
+    ) -> None:
+        super(GripNetInternalModule, self).__init__()
+        # in and out dimension
+        self.in_channels = in_channels
+        self.out_channels = setting.inter_agg_dim[-1]
+
+        self.num_edge_type = num_edge_type
+        self.if_multirelational = 1 if num_edge_type > 1 else 0
+        self.if_start_svertex = if_start_svertex
+        self.setting = setting
+
+        self.__init_inter_feat_layer__()
+        self.__init_inter_agg_layer__()
+
+    def __init_inter_feat_layer__(self):
+        """internal feature layer"""
+
+        self.embedding = torch.nn.Parameter(torch.Tensor(self.in_channels, self.setting.inter_feat_dim))
+        self.embedding.requires_grad = True
+
+        # reset parameters to be normally distributed
+        self.embedding.data.normal_()
+
+    def __init_inter_agg_layer__(self):
+        """internal aggregation layers"""
+
+        # compute the dim of input of the first internal aggregation layer
+        self.in_agg_dim = self.setting.inter_feat_dim
+        if not self.if_start_svertex:
+            assert self.setting.mode in [
```

Comment:
shall we change these `assert` here to `logging` as well?

Reply:
Changed in [9076b51](https://github.com/pykale/pykale/pull/338/commits/9076b511b7eacf20754998bb61e76b7f7c797e8f).